### DISCUSSION TITLE
prevent to jump back to the first page after editing a cell

### DISCUSSION THREAD

**Discussion by tttp** [KevinVandy/material-react-table#1020] [Q&A]
Hi,

my table has a few hundred rows and I enabled pagination 

    paginationDisplayMode: "pages",
    initialState: {
      pagination: { pageSize: 10, pageIndex: 0 },
    muiPaginationProps: {
      rowsPerPageOptions: [10, 50, 100],

I'm saving each cell change on blur, 

    onBlur: (event) => {
      const input = event.target;
      if (input.checkValidity) {
        if (!input.checkValidity()) {
          setError(cell.id, input.validationMessage); 
          return;
        }
        if (cell.id?.startsWith("mrt-row-create")) {
          setError(cell.id, ""); 
        }
        if (cell.id && !cell.id.startsWith("mrt-row-create_"))
          save(cell.id, event.target.value);
      }
    },

the save function does save on the server and onMutate modifies queryClient.setQueryData

it works fine when the edited cell is on the first page (first 10 rows) but if it's on the second page or after, the table resets and jumps back to the first page

Is there a way to prevent that? do I need to save the pagination state outside of material-react-table?

**Comment by AdmiJW** [comment]
Hi, see https://github.com/KevinVandy/material-react-table/discussions/552
